src/adapter/src/coord/message_handler.rs

```diff
@@ -275,10 +278,21 @@ impl Coordinator {
             });
         }
 
-        if let Err(err) = self.catalog_transact(None::<&Session>, ops).await {
-            tracing::warn!("Failed to update storage metrics: {:?}", err);
+        match self.catalog_transact_inner(None, ops).await {
+            Ok(table_updates) => {
+                let internal_cmd_tx = self.internal_cmd_tx.clone();
+                task::spawn(|| "storage_usage_update_table_updates", async move {
+                    table_updates
+                        .instrument(info_span!("coord::storage_usage_update::table_updates"))
```

Comment:
I copied and pasted this line and don't understand it at all. So please let me know if this looks wrong.

Reply:
This [appears](https://grafana.dev.materialize.com/explore?schemaVersion=1&panes=%7B%222qk%22:%7B%22datasource%22:%22c2718ff3-7d6c-493c-a9ec-214b5dd182ea%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22tempo%22,%22uid%22:%22c2718ff3-7d6c-493c-a9ec-214b5dd182ea%22%7D,%22queryType%22:%22traceqlSearch%22,%22limit%22:20,%22tableType%22:%22traces%22,%22filters%22:%5B%7B%22id%22:%228738c75f%22,%22operator%22:%22%3D%22,%22scope%22:%22span%22%7D,%7B%22id%22:%22service-name%22,%22tag%22:%22service.name%22,%22operator%22:%22%3D%22,%22scope%22:%22resource%22,%22value%22:%5B%22environmentd%22%5D,%22valueType%22:%22string%22%7D,%7B%22id%22:%22span-name%22,%22tag%22:%22name%22,%22operator%22:%22%3D%22,%22scope%22:%22span%22,%22value%22:%5B%22coord::storage_usage_update::table_updates%22%5D,%22valueType%22:%22string%22%7D%5D%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D,%22fk4%22:%7B%22datasource%22:%22c2718ff3-7d6c-493c-a9ec-214b5dd182ea%22,%22queries%22:%5B%7B%22query%22:%22c4a86ed9c72ccf5af8833942eeb1faf4%22,%22queryType%22:%22traceql%22,%22refId%22:%22A%22,%22limit%22:20,%22tableType%22:%22traces%22%7D%5D,%22range%22:%7B%22from%22:%221711054992809%22,%22to%22:%221711058592809%22%7D%7D%7D&orgId=1) as a new top-level trace with no parent spans. Is that what we'd expect from this change? I am not very familiar with this code, but [other spans](https://grafana.dev.materialize.com/explore?schemaVersion=1&panes=%7B%22zrt%22:%7B%22datasource%22:%22c2718ff3-7d6c-493c-a9ec-214b5dd182ea%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22tempo%22,%22uid%22:%22c2718ff3-7d6c-493c-a9ec-214b5dd182ea%22%7D,%22queryType%22:%22traceql%22,%22limit%22:20,%22tableType%22:%22traces%22,%22query%22:%22087e4b18a56f749940b0317f7500548a%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D%7D&orgId=1) I've looked at from this component had `coord::handle_message` and `storage_usage_update` etc as parent spans. Maybe @guswynn could help

Reply:
We probably want this to a otel child (not a tracing child) of `storage_usage_update`, so we want something like:
```
let mut task_span = info_span!(parent: None, "coord::storage_usage_update::table_updates");
OpenTelemetryContext::obtain().attach_as_parent_to(&mut task_span)
mz_ore::spawn(..., async move { ... }.instrument(task_span)
```

we should probably makes some macros to make this easier!

Reply:
Update, thanks!